### ISSUE TITLE
Disabling braintree breaks magento if v2.4.1 is installed?

### ISSUE THREAD

**Issue by zack6849** [dotmailer/dotmailer-magento2-extension#494]
Hi There, we recently updated your extension to v2.4.1in our development environment, and found out that if we disable the Magento_Braintree module, your extension breaks magento when we add things to cart, with the following stack trace:

```
Fatal error:  Uncaught Error: Class 'BraintreeFacade' not found in /home/acme/www_magento2/development/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:93
Stack trace:
#0 /home/acme/www_magento2/development/vendor/magento/framework/ObjectManager/Factory/Compiled.php(88): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('BraintreeFacade', Array)
#1 /home/acme/www_magento2/development/vendor/magento/framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Compiled->create('BraintreeFacade', Array)
#2 /home/acme/www_magento2/development/vendor/magento/module-payment/Model/Method/Factory.php(40): Magento\Framework\ObjectManager\ObjectManager->create('BraintreeFacade', Array)
#3 /home/acme/www_magento2/development/vendor/magento/module-payment/Helper/Data.php(112): Magento\Payment\Model\Method\Factory->create('BraintreeFacade')
#4 /home/acme/ in /home/acme/www_magento2/development/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php on line 93
```

I'm not really sure why that's happening, as I couldn't find any direct references to that class in your code, but nonetheless, disabling your module seems to reproducibly fix the problem

**Further troubleshooting information:**
```
Module Version: 2.4.1
Magento Version: 2.1.10
PHP Version: PHP 7.0.14
```

**Comment by cdiacon** [comment]
Hi @zack6849 we have found the issue and fill provide a fix soon. 

Thanks for reporting this.

**Comment by gorillagroup** [comment]
Then some module is disabled (and some of it classes is requred in another module) in app/etc/env.php  
this row in vendor/dotmailer/dotmailer-magento2-extension/etc/di.xml

<preference for="Magento\Framework\Module\ModuleListInterface" type="Magento\Framework\Module\FullModuleList" />

cause of issue.

**Comment by cdiacon** [comment]
Hi @zack6849 this has been fixed now in v2.4.3.
